src/org/etools/j1939_84/controllers/part06/Part06Step02Controller.java

```diff
@@ -54,9 +59,36 @@
     @Override
     protected void run() throws Throwable {
         // 6.6.2.1.a Global DM5 [(send Request (PGN 59904) for PGN 65230 (SPNs 1218-1219)]).
+        RequestResult<DM5DiagnosticReadinessPacket> globalDM5Result = getDiagnosticMessageModule().requestDM5(getListener());
+        List<DM5DiagnosticReadinessPacket> globalDM5Packets = globalDM5Result.getPackets();
+        List<DM5DiagnosticReadinessPacket> obdGlobalPackets = globalDM5Packets.stream()
+                                                                              .filter(p -> getDataRepository().isObdModule(p.getSourceAddress()))
+                                                                              .collect(Collectors.toList());
+
         // 6.6.2.2.a Fail if no OBD ECU reports a count of > 0 active DTCs.
+        boolean noActiveDTCs = obdGlobalPackets.stream().noneMatch(p -> (p.getActiveCodeCount() > 0));
+        if (noActiveDTCs) {
+            addFailure("6.6.2.2.a - No OBD ECU reported a count of > 0 active DTCs");
+        }
+
         // 6.6.2.2.b Fail if any OBD ECU reports > 0 previously active DTC.
+        obdGlobalPackets.stream()
+                        .filter(p -> p.getPreviouslyActiveCodeCount() != (byte) 0xFF
+                                && p.getPreviouslyActiveCodeCount() > 0)
+                        .map(ParsedPacket::getModuleName)
+                        .forEach(moduleName -> addFailure("6.6.2.2.b - OBD ECU " + moduleName
+                                + " reported > 0 previously active DTC"));
+
         // 6.6.2.3.a Warn if any ECU reports a count of > 1 active DTC or previously active DTC.
```

Comment:
Let's separate this warning into two parts so we can more clearly identify which count caused the warning.